### PR TITLE
Improve the warning message if the rbd command is not found.

### PR DESCRIPTION
**Pull request by sdkfzv**
**What this PR does / why we need it**:
The previous warning msg is not correct if there is no rbd cmd.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/volume/rbd/rbd_util.go [hunk 1]
```diff
@@ -137,9 +138,13 @@ func (util *RBDUtil) rbdLock(b rbdMounter, lock bool) error {
 		// for defencing, get the locker name, something like "client.1234"
 		cmd, err = b.plugin.execCommand("rbd",
 			append([]string{"lock", "list", b.Image, "--pool", b.Pool, "--id", b.Id, "-m", mon}, secret_opt...))
-		output = string(cmd)
-		glog.Infof("lock list output %q", output)
+		if cmd == nil {
```
Comment (krmayankk) [inline pkg/volume/rbd/rbd_util.go]: my understanding is that cmd will never be nil even if rbd is not found, isnt that so ?

Comment (sdkfzv) [inline pkg/volume/rbd/rbd_util.go]: Hi,
If the rbd is not found, the output is:
rbdLock **cmd: ""**; **err: executable file not found in $PATH**

From my tests, the 'cmd' will alway be nil if there is no rbd cmd. 
I'm not sure whether the 'err' will be that string or not and I'll do some more tests.
Thanks.

**Comment by sdkfzv** [comment]
Just updated the codes.

**Comment by spiffxp** [comment]
/ok-to-test
@kubernetes/sig-storage-pr-reviews (FYI @krmayankk using the github team to notify)

**Comment by sdkfzv** [comment]
I had signed the CLA, but how to run the checking again...

**Comment by sdkfzv** [comment]
Hi @justinsb @rootfs,
Could you kindly help to have a review, pls?
Thanks.

**Comment by sdkfzv** [comment]
I had signed the CLA

File: pkg/volume/rbd/rbd_util.go [hunk 2]
```diff
@@ -269,7 +270,11 @@ func (util *RBDUtil) AttachDisk(b rbdMounter) error {
 
 		// fence off other mappers
 		if err = util.fencing(b); err != nil {
-			return fmt.Errorf("rbd: image %s is locked by other nodes", b.Image)
+			if err.Error() == rbdCmdErr {
+				return fmt.Errorf("rbd: rbd cmd not found")
+			} else {
+				return fmt.Errorf("rbd: image %s is locked by other nodes", b.Image)
+			}
```
Comment (krmayankk) [inline pkg/volume/rbd/rbd_util.go]: nit: i would wrap this whole block into a function like `return rbdErrors()` or something and use it everywhere

Comment (sdkfzv) [inline pkg/volume/rbd/rbd_util.go]: The messages in the 'else' are different, so if we want to wrap it we need to make the function like **rbdErrors(err error, format string, args ...interface{})**. It seems complicated...
And sometimes we don't want to return it, such as line 447.
Thanks.

Comment (sdkfzv) [inline pkg/volume/rbd/rbd_util.go]: @krmayankk

**Comment by jsafrane** [comment]
please remove the useless comment and squash your comments so the PR can be merged, thanks!

**Comment by sdkfzv** [comment]
Hi @jsafrane ,
I think there is no other comments in the changeset need to be removed.
Please take a look.
Thanks.

**Comment by jsafrane** [comment]
oops, wrong PR, sorry :-)

File: pkg/volume/rbd/rbd_util.go (hunk 2 above)
Comment (rootfs) [inline pkg/volume/rbd/rbd_util.go]: a wrapper function like 
```go
func rbdErrors(runErr, resultErr,  error ) error {
 if err.Error() == rbdCmdErr {
   return fmt.Errorf("rbd: rbd cmd not found")
}
return resultErr
}
```

**Comment by sdkfzv** [comment]
I wrap the codes into a function rbdErrors().

**Comment by rootfs** [comment]
@sdkfzv please run hack/verify-gofmt.sh

**Comment by sdkfzv** [comment]
Hi @krmayankk  @rootfs ,
The checks have passed.
Could you take a look and what's the next step?
Thanks.

**Comment by supereagle** [comment]
@sdkfzv  No additional actions needed. This PR has been approved and will be automatically merged by k8s-merge-robot.

**Comment by sdkfzv** [comment]
@rootfs @justinsb,
Could you add the "lgtm label" if no other comments, pls?  Thanks.

**Pull request merged**